bcda/alr/csv/csv.go

```diff
@@ -0,0 +1,144 @@
+package csv
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/CMSgov/bcda-app/bcda/models"
+	"github.com/dimchansky/utfbom"
+	"github.com/go-gota/gota/dataframe"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	mbi   = "BENE_MBI_ID"
+	hic   = "BENE_HIC_NUM"
+	first = "BENE_1ST_NAME"
+	last  = "BENE_LAST_NAME"
+	sex   = "BENE_SEX_CD"
+	birth = "BENE_BRTH_DT"
+	death = "BENE_DEATH_DT"
+)
+
+// Fields the must be present in an ALR file
+var requiredFields []string = []string{
+	mbi, hic,
+	first, last,
+	sex,
+	birth, death,
+}
+
+// Fields that will be used to join multiple dataframes together
+var joinFields = requiredFields
+
+// ToALR reads in a CSV file(s) and unmarshals the data into an ALR model.
+// CSV files are joined based on a predetermined list of fields
+func ToALR(names ...string) ([]models.Alr, error) {
+	var mergedDF dataframe.DataFrame
+	for _, name := range names {
+		df, err := toDataFrame(name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create dataframe: %w", err)
+		}
+		if err := validate(df); err != nil {
+			return nil, fmt.Errorf("dataframe from %s is not valid: %w",
+				name, err)
+		}
+
+		if len(mergedDF.Names()) == 0 {
+			mergedDF = df
+		} else {
+			mergedDF = mergedDF.InnerJoin(df, joinFields...)
```

Comment:
We may have to react to cases like:
BENE_MBI_ID | BENE_HIC_NUM | BENE_1ST_NAME | BENE_LAST_NAME | BENE_SEX_CD | BENE_BRTH_DT | BENE_DEATH_DT
1-1
1234, 1234, John, Doe, Male, 1-15-1950, 
1-2
1234, <Missing>, John, Doe, Male, 1-15-1950, 

Or even
1-1
1234, 1234, John, Doe, Male, 1-15-1950, 
1-2
1234, 1234, John, Doe, M, 1-15-1950, 

Hopefully not.

Reply:
I wouldn't be surprised if we end up having to relax the joined columns so we only have MBI present. I'll put a comment in the code to describe the usage of the joinFields. 

For the Male -> M, I'm thinking the data dictionary should save us here. It actually calls out the expected values that we should have for each column. If we end up getting some unknown value, I think it makes sense to convey that when we encode it to the particular FHIR resource as UNKNOWN, not populate, etc.